lib/rational/src/Rational.cpp

```diff
@@ -33,24 +39,44 @@ Rational Rational::operator-(const Rational& other) {
 	a.numerator *= multiple / a.denominator;
 	b.numerator *= multiple / b.denominator;
 
-	return Rational(
+	Rational rez(
 		a.numerator - b.numerator,
 		multiple
-	).reduce();
+	);
+	rez.reduce();
+	this->numerator = rez.numerator;
+	this->denominator = rez.denominator;
+	
+	return rez;
 }
 
-Rational Rational::operator*(const Rational& other) {
-	return Rational(
+Rational& Rational::operator*=(const Rational& other) {
+	Rational rez(
 		this->numerator * other.numerator,
 		this->denominator * other.denominator
-	).reduce();
+	);
+	rez.reduce();
+	this->numerator = rez.numerator;
+	this->denominator = rez.denominator;
+	
+	return rez;
 }
 
-Rational Rational::operator/(const Rational& other) {
-	return Rational(
+Rational& Rational::operator/=(const Rational& other) {
+	Rational rez(
 		this->numerator * other.denominator,
 		this->denominator * other.numerator
-	).reduce();
+	);
+	rez.reduce();
+	this->numerator = rez.numerator;
+	this->denominator = rez.denominator;
+
+	return rez;
+}
+
+double Rational::toDouble() {
+	return static_cast<double> (numerator) /
+		static_cast<double> (denominator);
```

Comment:
use conversion to double with curly braces
double{numerator} / double{denominator}